extensions/shared/plan-utils.js

```diff
@@ -0,0 +1,122 @@
+
+/*
+ * Blocks list that require a paid plan.
+ */
+export const PAID_BLOCKS_LIST = [
+	// 'core/cover',
+	'core/video',
+];
+
+/*
+ * Blocks list that are free through FSE
+ */
+export const FREE_FSE_BLOCKS_LIST = [
+	'jetpack/layout-grid',
+	'jetpack/event-countdown',
+	'jetpack/timeline',
+	'jetpack/timeline-item',
+	'jetpack/field-email',
+];
+
+/**
+ * WP.com plan objects have a dedicated `path_slug` field,
+ * Jetpack plan objects don't.
+ * For Jetpack, we thus use the plan slug with the 'jetpack_' prefix removed.
+ *
+ * @param {object} PlanData -          Plan data object.
+ * @param {string} PlanData.planSlug - Plan slug.
+ * @param {object} PlanData.plan -     Object with details about the plan.
+ * @returns {string}                   Plan path slug.
+ */
+export function getPlanPathSlug( { planSlug, plan } ) {
+	return startsWith( planSlug, 'jetpack_' )
+		? planSlug.substr( 'jetpack_'.length )
+		: get( plan, [ 'path_slug' ] );
+}
+
+/**
+ * Return the checkout URL to upgrade the site plan,
+ * depending on the plan, postId, and postType site values.
+ *
+ * @param {object} siteParams -          Site params used to build the URL.
+ * @param {string} siteParams.planSlug - Plan slug.
+ * @param {string} siteParams.plan -     An object with details about the plan.
+ * @param {number} siteParams.postId -   Post id.
+ * @param {string} siteParams.postType - Post type.
+ * @returns {string}                     Upgrade URL.
+ */
+export function getUpgradeUrl( { planSlug, plan, postId, postType } ) {
+	const planPathSlug = getPlanPathSlug( { planSlug, plan } );
+
+	// The editor for CPTs has an `edit/` route fragment prefixed
+	const postTypeEditorRoutePrefix = [ 'page', 'post' ].includes( postType ) ? '' : 'edit';
+
+	// Post-checkout: redirect back here
+	const redirectTo = isSimpleSite()
+		? addQueryArgs(
+				'/' +
+					compact( [ postTypeEditorRoutePrefix, postType, getSiteFragment(), postId ] ).join( '/' ),
+				{
+					plan_upgraded: 1,
+				}
+		  )
+		: addQueryArgs(
+				window.location.protocol +
+					`//${ getSiteFragment().replace( '::', '/' ) }/wp-admin/post.php`,
+				{
+					action: 'edit',
+					post: postId,
+					plan_upgraded: 1,
+				}
+		  );
+
+	return (
+		planPathSlug &&
+		addQueryArgs( `https://wordpress.com/checkout/${ getSiteFragment() }/${ planPathSlug }`, {
+			redirect_to: redirectTo,
+		} )
+	);
+}
+
+/**
+ * Check if the block should is upgradable.
+ *
+ * @param {string} name - Block name.
+ * @returns {boolean} True if it should show the nudge. Otherwise, False.
+ */
+export function isUpgradable( name ) {
+	// Split up the block name to get blockNamespace/blockName.
+	const [ blockNamespace, blockName ] = /\//.test( name ) ? name.split( '/' ) : [ null, name ];
+
+	if (
+		blockNamespace &&
+		! [ 'jetpack', 'premium-content' ].includes( blockNamespace ) && // known namespaces.
+		! PAID_BLOCKS_LIST.includes( name )
+	) {
+		return false;
+	}
+
+	if ( FREE_FSE_BLOCKS_LIST.includes( name ) ) {
```

Comment:
These checks seem a bit odd. Shouldn't this function just check for the blocks that are tied to plans?

Reply:
This is something to research. Clearly hardcoding is not the best option. I think the idea is to expose the issue cc @cpapazoglou

Reply:
We just discovered (https://github.com/Automattic/jetpack/pull/16495#issuecomment-662395017) that some blocks provided by the FSE plugin are not included in the jetpack data https://github.com/Automattic/jetpack/blob/27afef4f91df4b99bb1d05a8e3a288c4fe5bc568/extensions/shared/get-jetpack-extension-availability.js#L19
thus they appear as paid. This is a temp solution while we are digging deeper right now

Reply:
This is now removed